### ISSUE TITLE
Quoted terms search returning docs it shouldn't in large docs

### ISSUE THREAD

**Issue by Sembiance** [typesense/typesense#779]
Searching for `"some word"` is returning docs that do not have those 2 words adjacent.

Download the attached [phraseMatchFiles.zip](https://github.com/typesense/typesense/files/9961701/phraseMatchFiles.zip) and unzip, start a server, create a scema and import a single document:
```
mkdir data
typesense-server --data-dir=./data --api-address=127.0.0.1 --api-port=8108 --peering-address=127.0.0.1 --api-key=TESTCASEKEY
curl "http://localhost:8108/collections" -X POST -H "Content-Type: application/json" -H "X-TYPESENSE-API-KEY: TESTCASEKEY" -d '{"name" : "twig", "fields" : [{ "name" : "content", "type" : "string", "optional" : true }] }'
curl "http://127.0.0.1:8108/collections/twig/documents/import" -X POST -H "X-TYPESENSE-API-KEY: TESTCASEKEY" --data-binary @"singledoc.jsonl"
```

Then perform a search:
```
curl --silent "http://127.0.0.1:8108/multi_search?query_by=content" -X POST -H "Content-Type: application/json" -H "X-TYPESENSE-API-KEY: TESTCASEKEY" --data-binary @"searchTest.json"
```

It returns 1 hit, but I was expecting ZERO hits, because in the content of the indexed document the words "frontier mail" do not appear together.

I had mentioned this on slack about 2 weeks ago and Kishore Nallan mentioned he was able to reproduce it and would investigate.

I just thought I should create a github issue for easier tracking of the issue.

**Comment by kishorenc** [comment]
This issue affects really large documents. We use uint16_t for storing the positions a particular token appears in a field. This limits us to accurately storing upto 64,000 positions. When a token index exceeds 64K, we will wrap around to 0. Since token positions are only used for calculating word proximity, this is mostly a good trade-off for the rare occasion that people store such large documents.

We had to recently refactor the code for supporting phrase search of repeating tokens, but this caused a regression where the new approach causes false positives due to the token index values wrapping around from 0 on large documents: two tokens that are not actually adjacent could have consecutive index values because of the wrap around.

I'm not quite sure how to fix this easily because it requires somehow knowing that a token offset is getting wrapped around, but we don't have this information easily accessible anywhere.

**Comment by kishorenc** [comment]
@Sembiance This is fixed in `typesense/typesense:0.24.0.rcn38` -- I have verified with the sample document you posted above as well.
